packages/vite-plugin-rsc/src/plugin.ts

```diff
@@ -0,0 +1,212 @@
+        }
+        config.build.rollupOptions ||= {};
+        config.build.rollupOptions.external = (id) => externals.has(id);
+      }
+    },
+
+    configResolved(resolvedConfig) {
+      config = resolvedConfig;
+      isSsr = !!config.build.ssr;
+      isBuild = config.command === "build";
+    },
+    resolveId(id, source) {
+      if (id === clientModuleId) {
+        return resolvedClientModuleId;
+      }
+      if (id === serverModuleId) {
+        return resolvedServerModuleId;
+      }
+    },
+    load(id) {
+      if (id === resolvedClientModuleId) {
+        return `export default ${JSON.stringify(
+          // Yes the client bundle map is in the server dist because
+          // it's the SSR build that generates the client bundle map
+          path.join(config.root || "", serverDist, clientBundleMapFilename)
+        )}`;
+      }
+      if (id === resolvedServerModuleId) {
+        return `export default ${JSON.stringify(
+          path.join(config.root || "", clientDist, serverBundleMapFilename)
+        )}`;
+      }
+    },
+    transform(code, id) {
+      // Short circuit if the file doesn't have the literal string
+      if (!code?.includes(clientPragma)) {
+        return;
+      }
+      // Check properly for the pragma
+      const ast = this.parse(code, { sourceType: "module" });
+      const localId = path.relative(config.root || "", id);
+      if (hasPragma(ast, clientPragma)) {
+        if (isSsr) {
+          const bundlePath = pathToFileURL(
+            path.join(config.root, clientDist, manifest[localId].file)
+          );
+          externals.add(bundlePath.href);
+          if (manifest[localId]) {
+            const exports = getExports(ast);
+            const exportProxies = exports
+              .map((name) => {
+                const symbolName = `${manifest[localId].file}#${name}`;
+                bundleMap.set(symbolName, {
+                  id: symbolName,
+                  chunks: [],
+                  name,
+                  async: true,
+                });
+                const localName = name === "default" ? "DefaultExport" : name;
+
+                return `
+                import { ${
+                  name === "default" ? "default as DefaultExport" : name
+                } } from ${JSON.stringify(
+                  bundlePath.href
+                )};${localName}.$$typeof = Symbol.for("react.client.reference");${localName}.$$id=${JSON.stringify(
+                  symbolName
+                )}; export ${
+                  name === "default" ? "default DefaultExport" : `{ ${name} }`
+                } `;
+              })
+              .join("\n");
+            return {
+              code: exportProxies,
+              map: { mappings: "" },
+            };
+          }
+        } else {
+          this.emitFile({
+            type: "chunk",
+            id,
+            preserveSignature: "allow-extension",
+          });
+        }
```

Comment:
If this is a client build, ensure that the client component is emitted in its own chunk, with export signatures retained so it can be imported manually in the client.